specs/deposit_gas_metering.md

```diff
@@ -0,0 +1,153 @@
+# Deposit Fee Spec
+
+<!-- START doctoc generated TOC please keep comment here to allow auto update -->
+<!-- DON'T EDIT THIS SECTION, INSTEAD RE-RUN doctoc TO UPDATE -->
+**Table of Contents**
+
+- [Deposits on L1](#deposits-on-l1)
+  - [Guaranteed Gas](#guaranteed-gas)
+  - [Gas Stipend](#gas-stipend)
+  - [Additional Gas](#additional-gas)
+  - [Limiting Guaranteed Gas](#limiting-guaranteed-gas)
+  - [Rationale for Guaranteed vs Additional Gas](#rationale-for-guaranteed-vs-additional-gas)
+  - [Rationale for burning L1 Gas](#rationale-for-burning-l1-gas)
+- [Deposits on L2](#deposits-on-l2)
+  - [Guaranteed Gas On L2](#guaranteed-gas-on-l2)
+  - [Additional Gas On L2](#additional-gas-on-l2)
+
+<!-- END doctoc generated TOC please keep comment here to allow auto update -->
+
+## Deposits on L1
+
+L1 Deposits are augmented with the following fields:
+
+- `guaranteedGas`
+- `gasPrice`
+- `additionalGas`
+
+`guaranteedGas` and `additionalGas` must fit into a 64 bit unsigned integer.  
+`gasPrice` must fit into a 256 bit unsigned integer.
+
+### Guaranteed Gas
+
+The **guaranteed gas** is composed of a gas stipend and any additional guaranteed
+gas the user would like to buy on L1.
```

Comment:
Since the term "stipend" has caused confusion in the past, I would precise here that the gas stipend is L2 gas given for free.